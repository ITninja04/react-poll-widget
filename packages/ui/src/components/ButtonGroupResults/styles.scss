@import "../../colors";

.button-group-results {
  position: relative;
  display: flex;
  flex-flow: row;
  width: 100%;
  background: $white;
  border-radius: 4px;
  font-size: 16px;
  font-weight: bold;
  overflow: hidden;

  .button-group-result {
    padding: 0.25em;
    flex: 1 1 auto;
    background: rgba($darkwhite, 0.8);
    
    &.winner {
      background: rgba($black, 0.4);
    }

    label {
      display: flex;
      flex-flow: column;

      span {
        text-align: center;
      }
    }
  }
}
