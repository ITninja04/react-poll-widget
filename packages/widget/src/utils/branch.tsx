export type BranchFunction = (condition: boolean, ComponentIfTrue: React.ReactNode, ComponentIfFalse: React.ReactNode) => React.ReactNode;

export const branch: BranchFunction = (condition, ComponentIfTrue, ComponentIfFalse) => condition
  ? ComponentIfTrue
  : ComponentIfFalse;