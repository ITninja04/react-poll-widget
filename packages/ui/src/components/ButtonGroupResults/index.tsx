import React from 'react';
import cx from 'classnames';
import map from 'lodash/map';
import max from 'lodash/max';

import './styles.scss';

import { PollResult } from '../types';
const ButtonGroupResults: React.FC<{
  results: PollResult[]
}> = ({
  results
}) => {
    const scores = map(results, result => result.score);
    const maxScore = max(scores);
    return (
      <div className='button-group-results'>
        {
          results.map(result => (
            <div
              key={result.option.id}
              className={cx(
                'button-group-result',
                { winner: maxScore === result.score }
              )}
              style={{ flex: `1 1 ${result.score}%` }}
            >
              <label>
                <span className='label'>{result.option.label}</span>
                <span className='score'>{result.score}%</span>
              </label>
            </div>
          ))
        }
      </div>
    );
  }

export default ButtonGroupResults;