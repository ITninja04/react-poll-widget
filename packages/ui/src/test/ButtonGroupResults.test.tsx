import ButtonGroupResults from "../components/ButtonGroupResults";
import { createSnapshotTest } from "./utils";

createSnapshotTest(ButtonGroupResults, {
  results: [
    { option: { id: '1', value: '1', label: 'Test 1' }, score: 25 },
    { option: { id: '2', value: '2', label: 'Test 2' }, score: 75 }
  ]
}, 'Right')

createSnapshotTest(ButtonGroupResults, {
  results: [
    { option: { id: '1', value: '1', label: 'Test 1' }, score: 66 },
    { option: { id: '2', value: '2', label: 'Test 2' }, score: 33 }
  ]
}, 'Left')