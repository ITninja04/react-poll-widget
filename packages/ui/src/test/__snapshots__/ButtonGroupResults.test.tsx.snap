// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Left should render 1`] = `
<DocumentFragment>
  <div
    class="button-group-results"
  >
    <div
      class="button-group-result winner"
      style="flex: 1 1 66%;"
    >
      <label>
        <span
          class="label"
        >
          Test 1
        </span>
        <span
          class="score"
        >
          66%
        </span>
      </label>
    </div>
    <div
      class="button-group-result"
      style="flex: 1 1 33%;"
    >
      <label>
        <span
          class="label"
        >
          Test 2
        </span>
        <span
          class="score"
        >
          33%
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Snapshot) Right should render 1`] = `
<DocumentFragment>
  <div
    class="button-group-results"
  >
    <div
      class="button-group-result"
      style="flex: 1 1 25%;"
    >
      <label>
        <span
          class="label"
        >
          Test 1
        </span>
        <span
          class="score"
        >
          25%
        </span>
      </label>
    </div>
    <div
      class="button-group-result winner"
      style="flex: 1 1 75%;"
    >
      <label>
        <span
          class="label"
        >
          Test 2
        </span>
        <span
          class="score"
        >
          75%
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
